### PR TITLE
Support of custom entrypoints

### PR DESCRIPTION
**Pull request by serhiicherepanov**
Hi, I found issue which happens if container use custom entrypoint. Bash commands cannot be executed. I move a CP command to overrided entrypoint. It's should help to use with customized entry points.

### CODE REVIEW

**Comment by Flink** [comment]
Thanks for your contribution! But I think it feels a little weird to have to rewrite the entrypoint of the image. I’ll make a patch to use the busybox image instead I think, so no problem and no need to add complex logic :)

**Comment by Flink** [comment]
Closed by 666c68e6f96f26823ff5cb086596e07e2c61a367

**Comment by serhiicherepanov** [comment]
@Flink no, it's needed for initial copy existing files from container to host. What will be copied if you use busybox?

**Comment by Flink** [comment]
Oops yes you’re right, don’t know what I was thinking yesterday… :confused:

File: post-build-dockerfile [hunk 1]
```diff
@@ -11,15 +11,29 @@ source "$(dirname "$0")/config"
 APP="$1"; APP_VOLUME="$VOLUMES_ROOT/$APP"; IMAGE="dokku/$APP"
 VOLUMES=$(docker inspect --format='{{range $volume, $value := .Config.Volumes}}{{$volume}}{{"\n"}}{{end}}' "$IMAGE")
 
-for VOLUME in $VOLUMES; do
-  VOLUME_PATH="$APP_VOLUME$VOLUME"
-  VOLUME_DIR=$(dirname "$VOLUME_PATH")
-  if [[ -e $VOLUME_PATH ]]; then
+for VOLUME in ${VOLUMES}; do
+  # Add slashes before spaces
+  VOLUME=$(echo ${VOLUME} | awk '{gsub(/ /,"\\ ")}8')
```
Comment (Flink) [inline post-build-dockerfile]: Are you sure this is necessary if all the variables are quoted?

File: post-build-dockerfile [hunk 2]
```diff
@@ -11,15 +11,29 @@ source "$(dirname "$0")/config"
...
+  VOLUME_PATH="${APP_VOLUME}${VOLUME}"
+  VOLUME_DIR=$(dirname "${VOLUME_PATH}")
+  if [[ -e ${VOLUME_PATH} ]]; then
     continue
   fi
-  mkdir -p "$VOLUME_DIR"
-  docker run --rm -v "$VOLUME_DIR:/target" "$IMAGE" cp -a "$VOLUME" /target
-  dokku docker-options:add "$APP" run,deploy "-v $VOLUME_PATH:$VOLUME"
+  dokku_log_info1 "VOLUME: Prepearing for: ${VOLUME}"
```
Comment (Flink) [inline post-build-dockerfile]: Could you remove calls to `dokku_log_*` please?

**Comment by Flink** [comment]
Could you quote all variables please instead of putting them between `{}`? Thanks!

File: post-build-dockerfile (hunk 1 above)
Comment (serhiicherepanov) [inline post-build-dockerfile]: It's needed for quote spaces in path, but i not sure this is best solution

File: post-build-dockerfile (hunk 2 above)
Comment (serhiicherepanov) [inline post-build-dockerfile]: dokku_log helps me to see more information about processes on server side at deploy.

**Comment by SergeyCherepanov** [comment]
I not a expert of bash scripting, but i found some issues with simple variable usage. Like this https://github.com/jansorg/bashsupport/issues/86, and as i know, it's not recommended to use.

**Comment by serhiicherepanov** [comment]
@Flink Updated!

**Pull request closed without merging**